src/coreclr/src/jit/simdashwintrinsic.cpp

```diff
@@ -1069,26 +1069,11 @@ GenTree* Compiler::impSimdAsHWIntrinsicCndSel(CORINFO_CLASS_HANDLE clsHnd,
     assert(op3 != nullptr);
 
 #if defined(TARGET_XARCH)
-    bool isVectorT256 = (simdSize == 32);
-
     // Vector<T> for the rel-ops covered here requires at least SSE2
     assert(compIsaSupportedDebugOnly(InstructionSet_SSE2));
 
     // Vector<T>, when 32-bytes, requires at least AVX2
-    assert(!isVectorT256 || compIsaSupportedDebugOnly(InstructionSet_AVX2));
-
-    if (compOpportunisticallyDependsOn(InstructionSet_SSE41))
```

Comment:
This was incorrect logic for conditional select as it was no longer doing a bitwise operation, which is what the current software fallback and SSE2 implementations are doing.